### DISCUSSION TITLE
Can't rebuild app, and the code can't file unknown file.

### DISCUSSION THREAD

**Discussion by ApinatXD** [invertase/react-native-firebase#8180] [Q&A] [answered]
When I run `./gradlew clean` in the `android `directory in order to delete the unused file before I rebuild the app, it gave this error:

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\Username\Appname\android\app\build.gradle' line: 175

* What went wrong:
Could not compile build file 'C:\Users\Usernane\Appname\android\app\build.gradle'.
> startup failed:
  build file 'C:\Users\Username\Appname\android\app\build.gradle': 175: only buildscript {}, pluginManagement {} and other plugins {} script blocks are allowed before plugins {} blocks, no other statements are allowed

  For more information on the plugins {} block, please refer to https://docs.gradle.org/8.11.1/userguide/plugins.html#sec:plugins_block in the Gradle documentation.

   @ line 175, column 1.
     plugins {
     ^

  1 error
```

And when I removed the plugin block to isolate the issue, the gradle code tries to find a `native_modules.graldle` file and I can't really find much information about it as it doesn't exist when I built my app.

Here's the error by the way:

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\Username\Appname\android\app\build.gradle' line: 172

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not read script 'C:\Users\Username\Appname\android\native_modules.gradle' as it does not exist.
```

I am really stuck with this and I cannot rebuild my app for my project to continue, any help would be appriciated :) 

P.S. The rebuild also won't work without the native_modules.gradle. To be honest, I don't have an idea what is the purpose of this file.

**Comment by mikehardy** [comment, marked as answer]
This is a project specific problem, I am able to use the gradle clean command in projects with react-native-firebase integrated and they work fine.

Did just now after creating a new project with my demo script https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

**Reply by ApinatXD** [reply]
Hi! Sorry the the late reply. I created this project a few months ago and it is fairly complex, to the point that I can't pinpoint the error, I just tried to delete that line of code and its been giving me other errors (i'll include it down below). I've been trying to find info about the native module file on the internet, but i can't find any.

```
Android Bundling failed 1224ms C:\Users\Username\Appname\index.js (1584 modules)
 ERROR  SyntaxError: C:\Users\Username\Appname\node_modules\react-native\Libraries\vendor\emitter\EventEmitter.js: Unexpected token, expected "]" (39:5)

  37 |
  38 | type Registry<TEventToArgsMap: {...}> = {
> 39 |   [K in keyof TEventToArgsMap]: Set<Registration<TEventToArgsMap[K]>>,
     |      ^
  40 | };
  41 |
  42 | /**
```

There's around 40++ error on this file, most of them is a syntax error.
